noir-projects/aztec-nr/aztec/src/utils/bytes.nr

```diff
@@ -0,0 +1,120 @@
+// Converts the input bytes into an array of fields. A Field is ~254 bits meaning that each field can store 31 bytes.
+// This implies that M = ceil(N / 31).
+pub fn bytes_to_fields<let N: u32, let M: u32>(input: [u8; N]) -> [Field; M] {
+    let mut dst = [0; M];
+
+    for field_index in 0..M {
+        let mut field_value = 0;
+
+        for i in 0..31 {
+            let byte_index = field_index * 31 + i;
+            if byte_index < N {
+                // Shift the existing value left by 8 bits and add the new byte
+                field_value = field_value * 256 + input[byte_index] as Field;
+            }
+        }
+
+        dst[field_index] = field_value;
+    }
+
+    dst
+}
+
+// Converts an input array of fields into bytes. Each field of input has to contain only 31 bytes.
+pub fn fields_to_bytes<let N: u32, let M: u32>(input: [Field; M]) -> [u8; N] {
+    let mut dst = [0; N];
+
+    for field_index in 0..M {
+        // We convert the field to 31 bytes. Normally a field is 254 bits, but we only used 248 bits
+        // in `bytes_to_fields` so we can safely convert it to 31 bytes.
```

Comment:
We could assert that `input[field_index] < 2^248` to detect incorrect usage.

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8590/commits/1e2c98deef79ff73050435edce173bc1937296ea